File: STP/src/gestionAplicacion/Personas/Conductor.java
package gestionAplicacion.Personas;
import java.util.ArrayList;

import gestionAplicacion.Vehiculos.*;


public class Conductor extends Persona {
	private Vehiculo vehiculo;
	private int experiencia;
	private double precio;
	public static ArrayList<Conductor> conductores = new ArrayList<Conductor>();
	
	//constructores
	public Conductor(String nombre, String documento, String direccion, String edad, Vehiculo vehiculo,
			int experiencia) {
		super(nombre, documento, direccion, edad);
		this.vehiculo = vehiculo;
		this.experiencia = experiencia;
	}
	public Conductor(String nombre, String documento,  String edad, 
			int experiencia,double precio) {
		super(nombre, documento, null, edad);
		this.experiencia = experiencia;
		this.precio = precio;
		Conductor.conductores.add(this);
	}

	public String presentacion() {
		return "Nombre: " + this.nombre + ". Edad: " + this.edad + " años. Experiencia: " + this.experiencia + " años"; 
	}
	//getters and setters
	public Vehiculo getVehiculo() {
		return vehiculo;
	}


	public void setVehiculo(Vehiculo vehiculo) {
		this.vehiculo = vehiculo;
	}


	public int getExperiencia() {
		return experiencia;
	}


	public void setExperiencia(int experiencia) {
		this.experiencia = experiencia;
	}
	
	public double getPrecio() {
		return precio;
	}
	public void setPrecio(double precio) {
		this.precio = precio;
	}
	//metodos 
	public void agregarVehiculo(Vehiculo v) {
		this.vehiculo = v;
		v.setConductor(this);
	}
	
	public static ArrayList<Conductor> getConductores(){
		return Conductor.conductores;
	}
}

File: STP-Python/src/gestionAplicacion/Personas/Conductor.py
from .Persona import Persona

class Conductor(Persona):
    conductores=[]
    
    def __init__(self, *args):
        if len(args) == 6:
            super().__init__(args[0], args[1], args[2], args[3])
            self._vehiculo = args[4]
            self._experiencia = args[5]
        if len(args) == 5:
            super().__init__(args[0], args[1], None, args[2])
            self._experiencia = args[3]
            self._precio = args[4]
            self.conductores.append(self)



    def setVehiculo (self,vehiculo):
        self._vehiculo = vehiculo

    def getVehiculo(self):
        return self._vehiculo

    def setExperiencia(self,experiencia):
        self._experiencia = experiencia

    def getExperiencia(self):
        return self._experiencia

    def setPrecio(self,precio):
        self._precio=precio
    
    def getPrecio(self):
        return self._precio

    @classmethod
    def setConductores(cls,conductores):
        cls.conductores = conductores

    @classmethod
    def getConductores(cls):
        return cls.conductores
    
    def presentacion(self):
        salida = self._nombre + ", " + str(self._edad) + " años. " + str(self._experiencia) + " años de experiencia"
        return salida 